src/app/clusters/thermostat-server/thermostat-server-presets.cpp

```diff
@@ -402,6 +402,24 @@ CHIP_ERROR ThermostatAttrAccess::AppendPendingPreset(Thermostat::Delegate * dele
         return CHIP_IM_GLOBAL_STATUS(ConstraintError);
     }
 
+    // Before adding this preset to the pending presets, if the expected length of the pending presets' list
+    // exceeds the total number of presets supported, return RESOURCE_EXHAUSTED. Note that the preset has not been appended yet.
+
+    uint8_t numberOfPendingPresets = CountNumberOfPendingPresets(delegate);
+
+    // Increment number of pending presets by 1 to account for this preset.
+    uint8_t totalExpectedCount = numberOfPendingPresets + 1;
+
+    uint8_t numberOfPresetsSupported = delegate->GetNumberOfPresets();
+
+    if (numberOfPresetsSupported == 0 || totalExpectedCount > numberOfPresetsSupported)
```

Comment:
Don't really need the first part, as totalExpectedCount is at least 1, but all this is likely to get overwritten when we do scenario checking anyway. I'll approve.

Reply:
done.